text/0083-starfish-tracing-model.md

```diff
@@ -0,0 +1,214 @@
+- Start Date: 2023-03-29
+- RFC Type: feature
+- RFC PR: [#83](https://github.com/getsentry/rfcs/pull/83)
+- RFC Status: draft
+
+# Summary
+
+This RFC proposes a new tracing model for Sentry's performance product to better
+address future product improvements.  The goal of this model is to allow
+**storing entire traces without gaps**, support **dynamic sampling**, **indexing
+of spans** and to **extract metrics from spans pre-sampling**.  This is an
+evolution of the current transaction based approach.
+
+# Motivation
+
+Today Sentry has a strong concept of a "transaction" which appears in multiple parts of the
+product.  It is both the transport of span data, the billable entity and the only indexed
+part of the product experience.  This means that spans that exist outside of the transaction
+cannot be represented and it also means that spans within a transaction are not indexed.
+
+The existing model has worked well for us to get started with evolving the Sentry errors
+product to capture performance traces, but it has restricted our ability to evolve the product
+forward.  It has created some restrictions on the SDK technology side (from high
+memory pressure, payload size limits) and also has promoted a separation of transactions and
+spans on the API layer which is untypical for tracing products.  It also has meant that Sentry
+has challenges with accepting traces coming directly from an OpenTelemetry exporter as the
+transaction concept is not a concept that OpenTelemetry has.
+
+We want to set a future direction that enables more flexible product choices and that we can
+move towards from our existing tracing model.  The goals are:
+
+* Support capturing entire traces
+* Have a data model story that allows us higher compatibility with Open Telemetry.  Specifically
+  we want a model that would permit us to ingest Open Telemetry data right from an exporter
+* Have a clear story for indexing and extracing metrics on a per-span level
+* Unified spans and transactions from an SDK perspective
+* Enable a path that allows clients to send 100% of spans outwards at least to a local aggregator,
+  preferrably a remote relay
```

Comment:
Which types of aggregation do you have in mind here ?
Is that the usual metrics extraction or something else?